contributing/code-style-and-quality-formatting.md

```diff
@@ -48,6 +48,45 @@ We are aware that spaces are a bit nicer; it just happened to be that we started
 * **Spaces around operators/keywords.** Operators (`+`, `=`, `>`, …) and keywords (`if`, `for`, `catch`, …) must have a space before and after them, provided they are not at the start or end of the line.
 
 
+### Breaking the lines of too long statements
+
+In general long lines should be avoided for the better readability. Try to use short statements which operate on the same level of abstraction. Break the long statements by creating more local variables, defining helper functions etc.
+
+Two major sources of long lines are:
+ * Long list of arguments in function declaration or call: <code>void func(type1 arg1, type2 arg2, ...)</code>
+ * Long sequence of chained calls: <code>list.stream().map(...).reduce(...).collect(...)...</code>
+
+Rules about breaking the long lines:
+ * Break the list if the line exceeds limit or earlier if you believe that the breaking would improve the code readability
+ * If you break the line then each argument/call should have a separate line, including the first one
+ * Each new line should have one extra indentation relative to the line of the parent function name or called entity
+
+Additionally for function arguments:
+ * The opening parenthesis always stays on the line of the parent function name
+ * The possible thrown exception list is never broken and stays on the same last line
```

Comment:
What if it conflicts with "Break the list if the line exceeds limit" when thrown exception list is too long?

Reply:
`break the list` was more meant about arguments or calls.
The mail discussion already mentions some arguments why not to break the exception list even if the line exceeds the length limit.
I can make it more detailed for clarity.